### ISSUE TITLE
Look into cargo-machete to replace cargo-udeps

### ISSUE THREAD

**Issue by syl20bnr** [tracel-ai/burn#2034]
From @Luni-4 on discord:

> we should replace cargo-udeps with cargo-machete which runs on stable and it's faster 

We use `cargo-udeps` in our `dependencies` xtask command here https://github.com/Tracel-AI/burn/blob/main/xtask/src/dependencies.rs#L88

This issue is about replacing it with [cargo-machete](https://github.com/bnjbvr/cargo-machete).

**Comment by tiruka** [comment]
@syl20bnr

Can I take this task if you do not mind?

Thank you.

**Comment by syl20bnr** [comment]
@tiruka Yes of course, but we are refactoring our xtask into a crate in the [tracel-xtask repository](https://github.com/tracel-ai/xtask). We are in the process of stabilizing the crate this week. I'll ping you next week when the tracel-xtask repo is ready to accept contributions.

**Comment by syl20bnr** [comment]
@tiruka The crates are released, you can start looking into it if you want.
